app/javascript/lib/sentry-logging.js

```diff
@@ -30,20 +44,41 @@ export function logJSFetchExceptionToSentry(response, titleInfo = '') {
  *
  * @param {Object} response - the response object from a failed JS fetch call
  * @param {String} titleInfo - extra info for the title of the Sentry event
+ * @param {Boolean} useThrottle - whether to apply clientside rate limit throttling
  */
-export function logJSFetchErrorToSentry(error, titleInfo = '') {
-  shouldLog() && Sentry.captureException(new Error(`${error}: ${titleInfo}`))
+export function logJSFetchErrorToSentry(error, titleInfo = '', useThrottle = false) {
+  if (shouldLog(useThrottle)) {
+    Sentry.captureException(new Error(`${error}: ${titleInfo}`))
+  }
+}
+
+/** Print suppression warning message to the console */
+function printSuppression(reason, response) {
+  if (isSuppressedEnv) {
+    reason = 'in an unlogged environment'
+  }
+  const message = `Suppressing error reporting to Sentry, because it is ${reason}.`
+  console.warn(response ? `${message }  Response:` : message)
+
+  // Error objects are printed via console.error already, so only surface Sentry-suppressed responses
+  if (response) {
+    console.warn(response)
+  }
 }
 
 /**
  * Determine if logging should occur based on environment
+ * @param {Boolean} useThrottle - whether to apply clientside rate limit throttling. Default false.
+ * @param {Object} response - Fetch response object.  Default null.
+ * @param {Number} sampleRate - % of events to log, only applied if `useThrottle = true`
+ *  1 = log all events, 0 = log no events, default = 0.05 (i.e. log 5% of events)
  */
-function shouldLog() {
-  const env = getSCPContext().environment
+export function shouldLog(useThrottle = false, response = null, sampleRate = 0.05) {
```

Comment:
BLOCKER: this seems to mix concerns, as this function does more than just make the determination, it also handles logging to console.  And it also has some logic that seems particular to only one sort of error with how it logs response errors.

  Let's rescope this function to  `logToSentry(errorMessage, useThrottle=false, throttleRate=0.05)`

Then this function can handle invoking `Sentry.captureException(errorMessage)` or logging that it was suppressed.  And it can return true/false on whether it was logged.  Then the callee (e.g. `logJSFetchExceptionToSentry`) can log anything else it wants (e.g. the response) if it finds the error was surpressed.  (or in this case, we might just want to log the response to console all the time)